File: backend/storage-service/src/Fix.Storage.Application/UseCases/Files/Dtos/FileTemplateDto.cs
using Fix.Storage.Domain.AggregateRoots.Files;

namespace Fix.Storage.Application.Files.Dtos;

public sealed record FileTemplateColumnDto(string Name, bool Required, string Description, string Example);

public sealed record FileTemplateDto(
    FileKind Kind,
    // Falso = só armazena.
    bool Processed,
    IReadOnlyList<string> AcceptedExtensions,
    IReadOnlyList<FileTemplateColumnDto> Columns,
    // CSV de exemplo em UTF-8 com BOM (vazio nos tipos só armazenados).
    byte[] ExampleCsv);

File: frontend/bff/src/application/files/dtos/file-template.dto.ts
import type { FileKind } from '../../../cross-cutting/enums/file-kind.js';

export interface FileTemplateColumnDto {
  name: string;
  required: boolean;
  description: string;
  example: string;
}

export interface FileTemplateDto {
  kind: FileKind;
  /** false = só armazena (políticas, documentos). */
  processed: boolean;
  /** Vazio = qualquer extensão. */
  acceptedExtensions: string[];
  columns: FileTemplateColumnDto[];
  /** CSV de exemplo (UTF-8 com BOM); vazio nos tipos só armazenados. */
  exampleCsv: Buffer;
}
